File: Medium/Dual Nim (TWONIM)/dual_nim.py
for _ in range(int(input())):
    n=int(input())
    a=list(map(int, input().split()))
    ans=0
    for i in a:
        ans^=i
    if(ans==0):
        print('First')
    else:
        if(n%2==0):
            print('First')
        else:
            print('Second')

File: Medium/Dual Nim (TWONIM)/dual_nim.java
import java.util.*;
import java .io.*;
public class Main
{
    public static void main(String[] args) throws IOException
    {
        Scanner in=new Scanner(System.in);
           int t=in.nextInt();
           while(t-->0){
           int n=in.nextInt();
           int[] a=new int[n];
           for(int i=0;i<n;i++)
           {
            
               a[i]=in.nextInt();
           }
          int ans=0;
           
            for(int i=0;i<n;i++)
           {
               ans^=a[i];
           }
           boolean a6=true;
           if(ans!=0)
           {
               if(n%2==1)
               {
                   a6=false;
                   
                   
                   
               }
           }
       
            System.out.println(a6?"First":"Second");  
           }
        }
       

    }

